packages/app/hooks/use-validation.tsx

```diff
@@ -0,0 +1,22 @@
+import { ValidationResult } from 'app/features/accounts/validators'
+import { useState } from 'react'
+
+export function useValidation<T>(
+  initialState: T,
+  validationFunction: (val: T) => ValidationResult
+) {
+  const [state, setState] = useState(initialState)
+  const [firstChangeLatch, setFirstChangeLatch] = useState(false)
```

Comment:
Hmm, name is a bit weird, also you should show errors in particular fields that show on either submit or blur and disappear once you change the value of the field again

Reply:
Perhaps best to use some established form validation library?

Reply:
https://blog.logrocket.com/react-form-validation-sollutions-ultimate-roundup/

Reply:
Started using Formik and Yup